### PR TITLE
add IRE purge

### PR DESCRIPTION
**Pull request by pbougue**
also secure lock release

### CODE REVIEW

**Comment by pbougue** [comment]
Jenkins, test this please

File: kirin/default_settings.py [hunk 1]
```diff
@@ -153,7 +153,17 @@
...
         'task': 'kirin.tasks.purge_gtfs_rt_update',
+        'schedule': schedules.crontab(hour='3', minute='15'),
+        'options': {'expires': 3600}
+    },
+    'purge_ire_trip_update': {
+        'task': 'kirin.tasks.purge_ire_trip_update',
         'schedule': schedules.crontab(hour='3', minute='30'),
         'options': {'expires': 3600}
+    },
+    'purge_ire_rt_update': {
```
Comment (pbougue) [inline kirin/default_settings.py]: We should comment for now it as discussed

**Comment by kadhikari** [comment]
jenkins test this

File: kirin/__init__.py [hunk 2]
```diff
@@ -67,6 +67,7 @@
 
 # activate a command
 import kirin.command.load_realtime
+import kirin.command.purge_rt
```
Comment (kinnou02) [inline kirin/__init__.py]: you can import it in the manage.py directly

File: kirin/tasks.py [hunk 3]
```diff
@@ -52,8 +63,54 @@ def close_session(*args, **kwargs):
...
+    lock_name = make_kirin_lock_name(func_name, connector)
+    with get_lock(logger, lock_name, app.config['REDIS_LOCK_TIMEOUT_PURGE']) as locked:
+        if not locked:
+            logger.warning('%s for %s is already in progress', func_name, connector)
+            return
+
+        until = datetime.date.today() - datetime.timedelta(days=int(config['nb_days_to_keep']))
+        logger.info('purge realtime update for {} until {}'.format(connector, until))
+
+        RealTimeUpdate.remove_by_connectors_until(connectors=[connector], until=until)
```
Comment (pbougue) [inline kirin/tasks.py]: please add a TODO in this method to ask for a param change: we want to purge on "contributor" later, not "connector".

File: kirin/__init__.py (hunk 2 above)
Comment (kinnou02) [reply kirin/__init__.py]: and you can also remove the old purge

**Pull request merged**